hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -862,11 +866,29 @@ public String startCommit(String actionType, HoodieTableMetaClient metaClient) {
     CleanerUtils.rollbackFailedWrites(config.getFailedWritesCleanPolicy(),
         HoodieTimeline.COMMIT_ACTION, () -> tableServiceClient.rollbackFailedWrites());
 
-    String instantTime = HoodieActiveTimeline.createNewInstantTime();
+    String instantTime = createCommit();
```

Comment:
Do we need this, even if there is rare oppotunity the instant time can conflict, one of the write would fail when trying to create the instant metadata file on the timeline, still workable?

Reply:
@danny0405, when using Spark insert overwrite historical partition and Flink insert into current partition, the different action type of same instant have no conflict and one of the write would fail for writing data file. We want that the behavior that insert overwrite historical partition doesn't affect the incremental writing tasks. Therefore we should create new and different instant time before starting commit to avoid various cases of boundaries due to the same instant time, for example, Spark insert overwrite historical partition and Flink insert into current partition.

Reply:
> that the behavior that insert overwrite historical partition doesn't affect the incremental writing tasks

Agree, but we also need to think through the trade-off, locking the instant time generation is kind of costly for most of the use cases, and we might encounter other issues because the file based lock can also time out in some use cases.

Reply:
Agree, the cost of lock is too high. when we fill back the history partitions in diff job, it will cost a lot of time to obtain it